styx-client/src/main/java/com/spotify/styx/client/auth/GoogleIdTokenAuth.java

```diff
@@ -0,0 +1,109 @@
+/*-
+ * -\-\-
+ * Spotify Styx API Client
+ * --
+ * Copyright (C) 2016 - 2017 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.styx.client.auth;
+
+import com.google.api.client.auth.oauth2.RefreshTokenRequest;
+import com.google.api.client.auth.oauth2.TokenRequest;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.util.Utils;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.webtoken.JsonWebSignature;
+import com.google.api.client.json.webtoken.JsonWebSignature.Header;
+import com.google.api.client.json.webtoken.JsonWebToken.Payload;
+import com.google.common.collect.ImmutableList;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+public class GoogleIdTokenAuth {
+  private final HttpTransport httpTransport;
+  private static final JsonFactory JSON_FACTORY = Utils.getDefaultJsonFactory();
+
+  GoogleIdTokenAuth(HttpTransport httpTransport) {
+    this.httpTransport = httpTransport;
+  }
+
+  public GoogleIdTokenAuth() {
+    this(Utils.getDefaultTransport());
+  }
+
+  public String getToken(String targetAudience) throws IOException, GeneralSecurityException {
```

Comment:
Seems the GCE metadata server hands out id tokens. Might be worth supporting. 

https://cloud.google.com/compute/docs/instances/verifying-instance-identity